Mathlib/MeasureTheory/Measure/CharacteristicFunction.lean

```diff
@@ -68,4 +91,104 @@ theorem ext_of_integral_char_eq (he : Continuous e) (he' : e ≠ 1)
 
 end ext
 
+section InnerProductSpace
+
+variable {E : Type*} {mE : MeasurableSpace E} {μ : Measure E} {t : E}
+
+/-- The characteristic function of a measure in an inner product space. -/
+noncomputable def charFun [Inner ℝ E] (μ : Measure E) (t : E) : ℂ := ∫ x, exp (⟪x, t⟫ * I) ∂μ
+
+lemma charFun_apply [Inner ℝ E] (t : E) : charFun μ t = ∫ x, exp (⟪x, t⟫ * I) ∂μ := rfl
+
+lemma charFun_apply_real {μ : Measure ℝ} (t : ℝ) :
+    charFun μ t = ∫ x, exp (t * x * I) ∂μ := by simp [charFun_apply]
+
+variable [SeminormedAddCommGroup E] [InnerProductSpace ℝ E]
+
+@[simp]
+lemma charFun_zero (μ : Measure E) : charFun μ 0 = (μ Set.univ).toReal := by
+  simp [charFun_apply]
+
+@[simp]
+lemma charFun_zero_measure : charFun (0 : Measure E) t = 0 := by simp [charFun_apply]
+
+@[simp]
+lemma charFun_neg (t : E) : charFun μ (-t) = conj (charFun μ t) := by
+  simp [charFun_apply, ← integral_conj, ← exp_conj]
+
+/-- `charFun` as the integral of a bounded continuous function. -/
+lemma charFun_eq_integral_innerProbChar : charFun μ t = ∫ v, innerProbChar t v ∂μ := by
+  simp [charFun_apply, innerProbChar_apply]
+
+lemma charFun_eq_integral_probChar (y : E) : charFun μ y = ∫ x, (probChar ⟪x, y⟫ : ℂ) ∂μ := by
+  simp [charFun_apply, probChar_apply]
+
+/-- `charFun` is a Fourier integral for the inner product and the character `probChar`. -/
+lemma charFun_eq_fourierIntegral (t : E) :
+    charFun μ t = VectorFourier.fourierIntegral probChar μ bilinFormOfRealInner 1 (-t) := by
+  simp [charFun_apply, VectorFourier.fourierIntegral_probChar]
+
+/-- `charFun` is a Fourier integral for the inner product and the character `fourierChar`. -/
+lemma charFun_eq_fourierIntegral' (t : E) :
+    charFun μ t
+      = VectorFourier.fourierIntegral fourierChar μ bilinFormOfRealInner 1 (-(2 * π)⁻¹ • t) := by
+  simp only [charFun_apply, VectorFourier.fourierIntegral, neg_smul,
+    bilinFormOfRealInner_apply_apply, inner_neg_right, inner_smul_right, neg_neg,
+    fourierChar_apply', Pi.ofNat_apply, Circle.smul_def, Circle.coe_exp, ofReal_mul, ofReal_ofNat,
+    ofReal_inv, smul_eq_mul, mul_one]
+  congr with x
+  rw [← mul_assoc, mul_inv_cancel₀ (by simp [pi_ne_zero]), one_mul]
+
+lemma norm_charFun_le (t : E) : ‖charFun μ t‖ ≤ (μ Set.univ).toReal := by
+  rw [charFun_eq_fourierIntegral]
+  exact (VectorFourier.norm_fourierIntegral_le_integral_norm _ _ _ _ _).trans_eq (by simp)
+
+lemma norm_charFun_le_one [IsProbabilityMeasure μ] (t : E) : ‖charFun μ t‖ ≤ 1 :=
+  (norm_charFun_le _).trans_eq (by simp)
+
+lemma norm_one_sub_charFun_le_two [IsProbabilityMeasure μ] : ‖1 - charFun μ t‖ ≤ 2 :=
+  calc ‖1 - charFun μ t‖
+  _ ≤ ‖(1 : ℂ)‖ + ‖charFun μ t‖ := norm_sub_le _ _
+  _ ≤ 1 + 1 := by simp [norm_charFun_le_one]
+  _ = 2 := by norm_num
+
+@[measurability]
```

Comment:
Is it useful to tag this one fun_prop too? (I don't know the answer!)

Reply:
`fun_prop` does not know about `StronglyMeasurable`.
We should change that, but that's for another PR!